### PR TITLE
Add correctness tests for String.GetHashCodeOrdinalIgnoreCase

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
This just tests the relationship between String.GetHashCode and String.GetHashCodeOrdinalIgnoreCase to ensure that we're not breaking the logic of one.

There will be a separate PR with Marvin-specific tests to validate the implementation.

### CODE REVIEW

File: src/System.Runtime/tests/System/StringGetHashCodeTests.cs [hunk 1]
```diff
@@ -60,5 +70,25 @@ public static IEnumerable<object[]> GetHashCode_TestData()
...
+                yield return new object[] { "abcdefgh".Substring(0, i) };
+                yield return new object[] { "ABCDEFGH".Substring(0, i) };
+            }
+
+            // 16 char mixed case mostly-ASCII string plus one non-ASCII character inserted at various locations
+            // tests fallback logic for OrdinalIgnoreCase hash
+
+            for (int i = 0; i <= 16; i++)
+            {
+                yield return new object[] { "AaBbCcDdEeFfGgHh".Insert(i, "\u00E9" /* LATIN SMALL LETTER E WITH ACUTE */) };
```
Comment (joshfree) [inline src/System.Runtime/tests/System/StringGetHashCodeTests.cs]: consider throwing in a cyrillic e and a turkish i for good measure

**Comment by GrabYourPitchforks** [comment]
@stephentoub Yup, the span-based methods would need equivalents.

File: src/System.Runtime/tests/System/StringGetHashCodeTests.cs (hunk 1 above)
Comment (Clockwork-Muse) [reply src/System.Runtime/tests/System/StringGetHashCodeTests.cs]: and non-ascii characters, with no upper/lower case equivalents, like 私は日本語を話します。(just a little)

**Comment by GrabYourPitchforks** [comment]
@dotnet-bot test Tizen armel Debug Build

**Comment by GrabYourPitchforks** [comment]
VS keeps adding UTF-8 BOMs to the .csproj files and it's making me sad. :(

**Pull request merged**